doc/nrf/libraries/caf/ble_smp.rst

```diff
@@ -38,19 +38,27 @@ The |smp| supports registering OS management handlers automatically, which you c
 Implementation details
 **********************
 
-The module registers the :c:func:`upload_confirm_cb` callback that is used to submit ``ble_smp_transfer_event``.
+The module periodically submits :c:struct:`ble_smp_transfer_event` during the image upload.
+The module registers the :c:func:`upload_confirm_cb` callback that is used to submit :c:struct:`ble_smp_transfer_event`.
 The module registers itself as the final subscriber of the event to track the number of submitted events.
-If a ``ble_smp_transfer_event`` was already submitted, but was not yet processed, the module desists from submitting subsequent event.
-After the previously submitted event is processed, the module submits a subsequent event when :c:func:`upload_confirm_cb` callback is called.
+If a :c:struct:`ble_smp_transfer_event` was already submitted but not processed, the module desists from submitting a subsequent event.
+After the previously submitted event is processed, the module submits a subsequent event when the :c:func:`upload_confirm_cb` callback is called.
 
 The application user must not perform more than one firmware upgrade at a time.
 The modification of the data by multiple application modules can result in a broken image that is going to be rejected by the bootloader.
 
-The module periodically submits ``ble_smp_transfer_event`` while the image is being uploaded.
-
-You can perform the DFU using for example the `nRF Connect for Mobile`_ application.
+You can perform the DFU using, for example, the `nRF Connect for Mobile`_ application.
 The :guilabel:`DFU` button appears in the tab with the connected Bluetooth devices.
 After pressing the button, you can select the :file:`*.bin` file that is to be used for the firmware update.
+Depending on the selected image slot either the :file:`zephyr/app_update.bin` or :file:`zephyr/mcuboot_secondary_app_update.bin` file must be used for the background DFU procedure.
+
+If the MCUboot bootloader uses the built-in direct-xip mode, you must upload the image to the slot that is currently unused.
+If the MCUboot bootloader uses the built-in swap mode instead, you must use the :file:`zephyr/app_update.bin` file.
+The :file:`zephyr/app_update.bin` file is the application image that can be booted from the primary slot.
```

Comment:
maybe it should be explained which file is for which slot right after line `"Depending on the selected image slot (...)"` and mention there `mcuboot_secondary_app_update.bin`